src/pip/_internal/utils/misc.py

```diff
@@ -181,8 +181,14 @@ def rmtree_errorhandler(func, path, exc_info):
     """On Windows, the files in .svn are read-only, so when rmtree() tries to
     remove them, an exception is thrown.  We catch that here, remove the
     read-only attribute, and hopefully continue without problems."""
+    try:
+        is_readonly = os.stat(path).st_mode & stat.S_IREAD
+    except (IOError, OSError):
+        # The path already removed, nothing to do
```

Comment:
We should have a comment saying that this is equivalent to `os.path.exists`, since at face value it looks overly broad.